onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -136,5 +136,68 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
   }
 }
 
+bool OnnxRuntimeTestSession::PopulateGeneratedInputTestData()
+{
+  size_t num_input_nodes;
+  num_input_nodes = session_.GetInputCount();
+
+  // iterate over all input nodes
+  for (size_t i = 0; i < num_input_nodes; i++) {
+    Ort::TypeInfo typeInfo = session_.GetInputTypeInfo(i);
+    Ort::MemoryInfo memory_info = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);
+    if (typeInfo.GetONNXType() == ONNX_TYPE_TENSOR) {
+        auto tensorInfo = typeInfo.GetTensorTypeAndShapeInfo();
+        std::vector<int64_t> input_node_dim = tensorInfo.GetShape();
+        int64_t input_tensor_size = 1;
+        for (auto dim : input_node_dim)
+        {
+          input_tensor_size *= dim;
+        }
+        size_t byteCount = input_tensor_size;
+        switch (tensorInfo.GetElementType())
+        {
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT:
+            byteCount *= sizeof(float);
+            break;
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT8:
+            byteCount *= sizeof(uint8_t);
+            break;
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT8:
+            byteCount *= sizeof(int8_t);
+            break;
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT16:
+            byteCount *= sizeof(uint16_t);
+            break;
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT16:
+            byteCount *= sizeof(int16_t);
+            break;
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT32:
+            byteCount *= sizeof(int32_t);
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64:
+            byteCount *= sizeof(int64_t);
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_BOOL:
+            byteCount *= sizeof(bool);
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_DOUBLE:
+            byteCount *= sizeof(double);
+            break;
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT32:
+            byteCount *= sizeof(uint32_t);
+            break;
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT64:
+            byteCount *= sizeof(uint64_t);
+            break;
+          default:
+            printf("Generation of input for this element type not implemented.");
+            return false;
+        }
+        void* input_tensor_values = malloc(byteCount);
+        generated_input_buffers_.push_back(input_tensor_values);
```

Comment:
if generated_input_buffers_ were a `std::vector<std::vector<int8_t>> generated_input_buffers_`, you could replace these lines with `generated_input_buffers_.emplace_back(byteCount)` and enjoy RAII by removing the explicit `free` calls later on :)

Reply:
I'm not so sure how this works 😅

Reply:
Nevermind, I'm migrating away from this malloc approach :)